controllers/clustermodule_reconciler.go

```diff
@@ -253,3 +253,29 @@ func (r Reconciler) fetchMachineOwnerObjects(ctx *context.ClusterContext) (map[s
 func appendKCPKey(name string) string {
 	return "kcp" + name
 }
+
+func incompatibleOwnerErrors(errList []clusterModError) []clusterModError {
+	toReport := []clusterModError{}
+	for _, e := range errList {
+		if clustermodule.IsIncompatibleOwnerError(e.err) {
+			toReport = append(toReport, e)
+		}
+	}
+	return toReport
+}
+
+type clusterModError struct {
+	name string
+	err  error
+}
+
+func generateClusterModuleErrorMessage(errList []clusterModError) string {
+	sb := strings.Builder{}
+	sb.WriteString("failed to create cluster modules for: ")
+
+	for _, e := range errList {
+		sb.WriteString(fmt.Sprintf("%s %s, ", e.name, e.err.Error()))
+	}
+	msg := sb.String()
+	return msg[:len(msg)-2]
```

Comment:
why is this len -2 ?

Reply:
Remove the trailing `, ` after the final object name and error